tests/test_nhg.py

```diff
@@ -216,6 +287,358 @@ def test_route_nhg(self, dvs, dvs_route, testlog):
+        nhg_ps._del("group1")
+        time.sleep(1)
+        assert len(get_nhg_keys()) == asic_nhgs_count
+
+        # Create a next hop group with a member that does not exist - should fail
+        fvs = swsscommon.FieldValuePairs([('nexthop', '10.0.0.1,10.0.0.3,10.0.0.63'),
+                                        ("ifname", "Ethernet0,Ethernet4,Ethernet124")])
+        nhg_ps.set("group1", fvs)
+        time.sleep(1)
+        assert len(asic_db.get_keys(self.ASIC_NHG_STR)) == asic_nhgs_count
+
+        # Issue an update for this next hop group that doesn't yet exist,
+        # which contains only valid NHs.  This will overwrite the previous
+        # operation and create the group.
+        fvs = swsscommon.FieldValuePairs([('nexthop', '10.0.0.1,10.0.0.5'),
+                                        ("ifname", "Ethernet0,Ethernet8")])
+        nhg_ps.set("group1", fvs)
+        asic_db.wait_for_n_keys(self.ASIC_NHG_STR, asic_nhgs_count + 1)
+        asic_db.wait_for_n_keys(self.ASIC_NHGM_STR, asic_nhgms_count + 2)
+
+        # Check the group has it's two members
+        for nhgid in asic_db.get_keys(self.ASIC_NHG_STR):
+            if nhgid not in prev_nhg_keys:
+                break
+
+        count = 0
+        for k in asic_db.get_keys(self.ASIC_NHGM_STR):
+            fvs = asic_db.get_entry(self.ASIC_NHGM_STR, k)
+            if fvs['SAI_NEXT_HOP_GROUP_MEMBER_ATTR_NEXT_HOP_GROUP_ID'] == nhgid:
+                count += 1
+        assert count == 2
+
+        # Add a route referencing the new group
+        asic_rts_count = len(get_rt_keys())
+        fvs = swsscommon.FieldValuePairs([('nexthop_group', 'group1')])
+        rt_ps.set('2.2.2.0/24', fvs)
+        asic_db.wait_for_n_keys(self.ASIC_RT_STR, asic_rts_count + 1)
+
+        # Get the route key
+        for rt_key in get_rt_keys():
+            k = json.loads(rt_key)
+
+            if k['dest'] == "2.2.2.0/24":
+                break
+
+        # Try removing the group while it still has references - should fail
+        nhg_ps._del('group1')
+        time.sleep(1)
+        assert len(get_nhg_keys()) == asic_nhgs_count + 1
+
+        # Create a new group
+        fvs = swsscommon.FieldValuePairs([('nexthop', '10.0.0.1,10.0.0.3'),
+                                          ('ifname', 'Ethernet0,Ethernet4')])
+        nhg_ps.set("group2", fvs)
+        asic_db.wait_for_n_keys(self.ASIC_NHG_STR, asic_nhgs_count + 2)
+        asic_db.wait_for_n_keys(self.ASIC_NHGM_STR, asic_nhgms_count + 4)
+
+        # Update the route to point to the new group
+        fvs = swsscommon.FieldValuePairs([('nexthop_group', 'group2')])
+        rt_ps.set('2.2.2.0/24', fvs)
+
+        # The first group should have got deleted
+        asic_db.wait_for_n_keys(self.ASIC_NHG_STR, asic_nhgs_count + 1)
+        asic_db.wait_for_n_keys(self.ASIC_NHGM_STR, asic_nhgms_count + 2)
+
+        assert asic_db.get_entry(self.ASIC_RT_STR, rt_key)['SAI_ROUTE_ENTRY_ATTR_NEXT_HOP_ID'] != nhgid
```

Comment:
can you add a comment here since the check is being done here. Any time it fails it will be easier for the user to debug.

Reply:
Done